SimCalorimetry/EcalSimAlgos/src/EcalLiteDTUCoder.cc

```diff
@@ -0,0 +1,241 @@
+#include "SimCalorimetry/EcalSimAlgos/interface/EcalLiteDTUCoder.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "SimGeneral/NoiseGenerators/interface/CorrelatedNoisifier.h"
+#include "DataFormats/EcalDigi/interface/EcalLiteDTUSample.h"
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+
+#include "CondFormats/EcalObjects/interface/EcalConstants.h"
+
+#include <iostream>
+
+//#include "Geometry/CaloGeometry/interface/CaloGenericDetId.h"
+
+/////////////////////
+//NOTE:
+//TO BE CORRECTLY IMPLEMENTED!
+//Simona Cometti, [20.03.20 12:05]
+//[In reply to Dario Soldi]
+//Allora nella versione 1 i samples a gain 1 erano solo 8
+//Centrati così:
+//Quindi 2 samples prima e 5 dopo a quello saturato
+//Nella versione 1.2 c'è la possibilità di avere 8 samples o 16 samples dal gain 1
+//Quando è selezionata l'opzione dei 16 samples:
+//5 samples prima, il sample saturo e poi 10 samples dopo
+const float DTUVersion = 1.0;
+
+EcalLiteDTUCoder::EcalLiteDTUCoder(bool addNoise,
+                             bool PreMix1,
+                             EcalLiteDTUCoder::Noisifier* ebCorrNoise0,
+                             EcalLiteDTUCoder::Noisifier* ebCorrNoise1)
+    : m_peds(nullptr),
+      m_gainRatios(nullptr),
+      m_intercals(nullptr),
+      //   m_maxEneEB    (      1668.3 ) , // 4095(MAXADC)*12(gain 2)*0.035(GeVtoADC)*0.97
+      m_maxEneEB(ecalPh2::maxEneEB),  // Maximum for CATIA: LSB gain 10: 0.048 MeV
+      m_addNoise(addNoise),
+      m_PreMix1(PreMix1)
+
+{
+  m_ebCorrNoise[0] = ebCorrNoise0;
+  assert(nullptr != m_ebCorrNoise[0]);
```

Comment:
is this possibility expected to occur in production? if it was just a development check, remove it; otherwise, make it an exception with a clear message